src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy

```diff
@@ -0,0 +1,520 @@
+
+        then: "PBs shouldn't emit log"
+        def logsByTime = pbsServiceWithResponseCorrectionModule.getLogsByTime(start)
+        def responseCorrection = getLogsByText(logsByTime, bidResponse.seatbid[0].bid[0].id)
+        assert responseCorrection.size() == 0
+
+        and: "Response should contain seatBid type"
+        assert response.seatbid.size() == 1
+        assert response.seatbid[0].bid[0].ext.prebid.type == mediaType
+        assert !response?.seatbid[0]?.bid[0]?.ext?.prebid?.meta?.mediaType
+
+        and: "Response shouldn't contain errors"
+        assert !response.ext.errors
+
+        and: "Response shouldn't contain warnings"
+        assert !response.ext.warnings
+
+        where:
+        mediaType << [BANNER, AUDIO]
+    }
+
+    def "PBs shouldn't modify response and emit logs when requested impression with native and adm value is asset"() {
+        given: "Start up time"
+        def start = Instant.now()
+
+        and: "Default bid request with APP and #mediaType imp"
+        def bidRequest = getDefaultBidRequest(APP).tap {
+            imp[0] = Imp.getDefaultImpression(NATIVE)
+        }
+
+        and: "Set bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        def bidId = bidResponse.seatbid[0].bid[0].id
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        and: "Save account with enabled response correction module"
+        accountDao.save(accountConfigWithResponseCorrectionModule(bidRequest))
+
+        when: "PBS processes auction request"
+        def response = pbsServiceWithResponseCorrectionModule.sendAuctionRequest(bidRequest)
+
+        then: "PBs shouldn't emit log"
+        def logsByTime = pbsServiceWithResponseCorrectionModule.getLogsByTime(start)
+        def responseCorrection = getLogsByText(logsByTime, bidId)
+        assert responseCorrection[0].contains("Bid $bidId of bidder generic has an JSON ADM, that appears to be native" as String)
+        assert responseCorrection.size() == 1
+
+        and: "Response should contain seatBid type"
+        assert response.seatbid.size() == 1
+        assert response.seatbid[0].bid[0].ext.prebid.type == NATIVE
+
+        and: "Response shouldn't contain errors"
+        assert !response.ext.errors
+
+        and: "Response shouldn't contain warnings"
+        assert !response.ext.warnings
+    }
+
+    def "PBs shouldn't modify response when requested video impression respond with empty adm"() {
+        given: "Start up time"
+        def start = Instant.now()
+
+        and: "Default bid request with APP and Video imp"
+        def bidRequest = getDefaultBidRequest(APP).tap {
+            imp[0] = Imp.getDefaultImpression(VIDEO)
+        }
+
+        and: "Set bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid[0].bid[0].setAdm(null)
+            seatbid[0].bid[0].nurl = PBSUtils.randomString
```

Comment:
What purpose of `nurl` in this test?

Reply:
ResponseBidValidator:131L -> validateTypeSpecific(...)